src/ai/backend/manager/models/user.py

```diff
@@ -666,6 +679,246 @@ async def mutate(cls, root, info, email):
                 return cls(ok=False, msg=f'unexpected error: {e}')
 
 
+class PurgeUser(graphene.Mutation):
+    '''
+    Delete user as well as all user-related DB informations such as keypairs, kernels, etc.
+
+    If target user has virtual folders, they can be purged together or migrated to the superadmin.
+
+    vFolder treatment policy:
+      User-type:
+      - vfolder is not shared: delete
+      - vfolder is shared:
+        + if purge_shared_vfolder is True: delete
+        + else: change vfolder's owner to requested admin
+
+    This action cannot be undone.
+    '''
+    allowed_roles = (UserRole.SUPERADMIN,)
+
+    class Arguments:
+        email = graphene.String(required=True)
+        props = PurgeUserInput(required=True)
+
+    ok = graphene.Boolean()
+    msg = graphene.String()
+
+    @classmethod
+    async def mutate(cls, root, info, email, props):
+        purge_shared_vfolders = props.purge_shared_vfolders if props.purge_shared_vfolders else False
+        async with info.context['dbpool'].acquire() as conn, conn.begin():
+            try:
+                query = (
+                    sa.select([users.c.uuid])
+                    .select_from(users)
+                    .where(users.c.email == email)
+                )
+                user_uuid = await conn.scalar(query)
+                log.info('completly deleting user {0}...', email)
+
+                if not purge_shared_vfolders:
+                    await cls.migrate_shared_vfolders(
+                        conn,
+                        deleted_user_uuid=user_uuid,
+                        target_user_uuid=info.context['user']['uuid'],
+                        target_user_email=info.context['user']['email'],
+                    )
+                await cls.delete_vfolders(conn, user_uuid, info.context['config_server'])
+                await cls.delete_kernels(conn, user_uuid)
```

Comment:
We need to block these operations if there are non-terminated kernels or vfolders mounted onto running containers.